kroxylicious-api/src/main/java/io/kroxylicious/proxy/config/tls/Tls.java

```diff
@@ -8,17 +8,42 @@
 
 import java.security.KeyStore;
 import java.util.Locale;
+import java.util.Objects;
+
+import io.kroxylicious.proxy.tag.VisibleForTesting;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
 
 /**
  * Provides TLS configuration for this peer.  This class is designed to be used for both TLS server and client roles.
  *
  * @param key   specifies a key provider that provides the certificate/key used to identify this peer.
  * @param trust specifies a trust provider used by this peer to determine whether to trust the peer. If omitted platform trust is used instead.
+ * @param clientAuth specifies how the TLS session should authenticate clients. Optional not required when configuring upstream TLS
  *
  * TODO ability to restrict by TLS protocol and cipher suite.
  */
 public record Tls(KeyProvider key,
-                  TrustProvider trust) {
+                  TrustProvider trust,
+                  @Nullable TlsClientAuth clientAuth) {
+
+    // Sonar is wrong about this being equivalent to the default ctor as the fields are not initialised by the default before calling the validation.
+    @SuppressWarnings("java:S6207")
+    public Tls(KeyProvider key, TrustProvider trust, TlsClientAuth clientAuth) {
+        this.key = key;
+        this.trust = trust;
+        this.clientAuth = clientAuth;
+        validateClientAuth();
+    }
+
+    /**
+     * @deprecated use the all args constructor {@see io.kroxylicious.proxy.config.tls.Tls#Tls(io.kroxylicious.proxy.config.tls.KeyProvider, io.kroxylicious.proxy.config.tls.TrustProvider, io.kroxylicious.proxy.config.tls.TlsClientAuth)}
+     */
+    // This is required for API backwards compatability
+    @Deprecated(forRemoval = true, since = "0.10.0")
```

Comment:
Hmmm. Thinking about this. Should we really deprecate and remove this signature given its used for upstream connections where client auth isn't currently supported?

Reply:
Agreed, that makes sense to me.